trace-normalization/benches/normalization_utils.rs

```diff
@@ -15,50 +21,63 @@ fn normalize_service_bench(c: &mut Criterion) {
             "A00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 000000000000",
         ];
 
-    for case in cases {
-        group.bench_with_input(
-            BenchmarkId::new(
-                "normalize_service",
-                if case.is_empty() {
-                    "[empty string]"
-                } else {
-                    case
-                },
-            ),
-            *case,
-            |b, case| {
-                b.iter_batched_ref(
-                    || case.to_owned(),
-                    datadog_trace_normalization::normalize_utils::normalize_service,
-                    BatchSize::NumBatches(100000),
-                )
-            },
-        );
-    }
-    group.finish()
+    normalize_fnmut_string(group, cases, 500, "normalize_service", normalize_service);
 }
 
 fn normalize_name_bench(c: &mut Criterion) {
-    let mut group = c.benchmark_group("normalization/normalize_name");
+    let group = c.benchmark_group("normalization/normalize_name");
     let cases = &[
         "good",
         "bad-name",
         "Too-Long-.Too-Long-.Too-Long-.Too-Long-.Too-Long-.Too-Long-.Too-Long-.Too-Long-.Too-Long-.Too-Long-.Too-Long-.",
     ];
+    normalize_fnmut_string(group, cases, 500, "normalize_name", normalize_name);
+}
+
+#[inline]
+fn normalize_fnmut_string<F>(
+    mut group: BenchmarkGroup<WallTime>,
+    cases: &[&str],
+    elements: u64,
+    function_name: &str,
+    mut function: F,
+) where
+    F: FnMut(&mut String),
+{
+    // Measure over a number of calls to minimize impact of OS noise
+    group.throughput(Elements(elements));
+    // We only need to measure for a small time since the function is very fast
+    group.warm_up_time(Duration::from_secs(1));
+    group.measurement_time(Duration::from_secs(3));
+    group.sample_size(200);
+    group.sampling_mode(criterion::SamplingMode::Flat);
+
     for case in cases {
         group.bench_with_input(
-            BenchmarkId::new("normalize_name", case),
+            BenchmarkId::new(
+                function_name,
+                if case.is_empty() {
+                    "[empty string]"
+                } else {
+                    case
+                },
+            ),
             *case,
             |b, case| {
                 b.iter_batched_ref(
                     || case.to_owned(),
-                    datadog_trace_normalization::normalize_utils::normalize_name,
-                    BatchSize::NumIterations(100000),
+                    |mutable| {
+                        (0..elements).for_each(|_| {
+                            mutable.insert_str(0, case);
```

Comment:
This is not ideal, having a memcopy in the measured part, but we need to measure over multiple calls to minimize the noise 🤷🏼‍♂️

Reply:
I'm fixing this to allocate the Strings up front.